HADOOP-19731. Fix SpotBugs warnings introduced after SpotBugs version upgrade.

zhtttylz opened the pull request:
### Description of PR
HADOOP-19731. Fix SpotBugs warnings introduced after SpotBugs version upgrade.

### How was this patch tested?

Ran `mvn -Dspotbugs.skip=false spotbugs:spotbugs` on affected modules and verified the build no longer fails on SpotBugs warnings. No functional code changes, config-only.

### For code changes:

- Add a project-wide baseline at dev-support/findbugs-exclude-global.xml. 
- Consolidate SpotBugs plugin config in affected module POMs to consistently include local excludes and the new global baseline. 
- Wire the global baseline from hadoop-project-dist/pom.xml; introduce a root path property to reference the repository root.

slfan1989 commented:
@zhtttylz Thank you for following up on this issue.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 29s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  1s |  |  xmllint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   8m 39s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  15m 41s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   8m 19s |  |  trunk passed with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  compile  |   8m 22s |  |  trunk passed with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04  |
| -1 :x: |  mvnsite  |   6m 23s | [/branch-mvnsite-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/2/artifact/out/branch-mvnsite-root.txt) |  root in trunk failed.  |
| +1 :green_heart: |  javadoc  |   5m 21s |  |  trunk passed with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   4m 44s |  |  trunk passed with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  shadedclient  |  71m 43s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 24s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |  25m 51s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   8m  1s |  |  the patch passed with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |   8m  1s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   8m 20s |  |  the patch passed with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |   8m 20s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -1 :x: |  mvnsite  |   3m 41s | [/patch-mvnsite-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/2/artifact/out/patch-mvnsite-root.txt) |  root in the patch failed.  |
| +1 :green_heart: |  javadoc  |   5m 15s |  |  the patch passed with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   4m 38s |  |  the patch passed with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  shadedclient  |  28m 29s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 589m 16s | [/patch-unit-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/2/artifact/out/patch-unit-root.txt) |  root in the patch passed.  |
| -1 :x: |  asflicense  |   0m 48s | [/results-asflicense.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/2/artifact/out/results-asflicense.txt) |  The patch generated 1 ASF License warnings.  |
|  |   | 734m 52s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.yarn.server.router.subcluster.capacity.TestYarnFederationWithCapacityScheduler |
|   | hadoop.security.ssl.TestDelegatingSSLSocketFactory |
|   | hadoop.yarn.sls.appmaster.TestAMSimulator |
|   | hadoop.hdfs.TestDecommission |
|   | hadoop.hdfs.tools.TestDFSAdmin |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.51 ServerAPI=1.51 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/8053 |
| Optional Tests | dupname asflicense codespell detsecrets xmllint compile javac javadoc mvninstall mvnsite unit shadedclient |
| uname | Linux 2197de4c49c9 5.15.0-156-generic #166-Ubuntu SMP Sat Aug 9 00:02:46 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / c46c6afc99a346474f5b255749c5a86ae4de90bc |
| Default Java | Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04 |
| Multi-JDK versions | /usr/lib/jvm/java-21-openjdk-amd64:Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04 /usr/lib/jvm/java-17-openjdk-amd64:Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/2/testReport/ |
| Max. process+thread count | 4391 (vs. ulimit of 5500) |
| modules | C: hadoop-project-dist hadoop-common-project/hadoop-minikdc hadoop-common-project/hadoop-auth hadoop-common-project/hadoop-nfs hadoop-common-project/hadoop-kms hadoop-common-project/hadoop-registry hadoop-hdfs-project/hadoop-hdfs-httpfs hadoop-yarn-project/hadoop-yarn hadoop-mapreduce-project/hadoop-mapreduce-client hadoop-mapreduce-project/hadoop-mapreduce-examples hadoop-mapreduce-project hadoop-tools/hadoop-streaming hadoop-tools/hadoop-archive-logs hadoop-tools/hadoop-rumen hadoop-tools/hadoop-gridmix hadoop-tools/hadoop-datajoin hadoop-tools/hadoop-aws hadoop-tools/hadoop-azure hadoop-tools/hadoop-aliyun hadoop-tools/hadoop-sls hadoop-tools/hadoop-fs2img hadoop-tools/hadoop-gcp hadoop-tools/hadoop-benchmark hadoop-cloud-storage-project/hadoop-cos hadoop-cloud-storage-project/hadoop-huaweicloud hadoop-cloud-storage-project/hadoop-tos . U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/2/console |
| versions | git=2.25.1 maven=3.9.11 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m  0s |  |  Docker mode activated.  |
| -1 :x: |  docker  |  14m 34s |  |  Docker failed to build run-specific yetus/hadoop:tp-29464}.  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hadoop/pull/8053 |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/4/console |
| versions | git=2.34.1 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

slfan1989 commented:
@steveloughran @szetszwo We propose adding a global configuration to temporarily suppress the warnings generated by the new rules introduced in SpotBugs 4.9.7. Would this approach be acceptable to you?

szetszwo commented:
> ...  adding a global configuration to temporarily suppress the warnings generated by the new rules introduced in SpotBugs 4.9.7.  ...

@slfan1989, @zhtttylz  , it sounds good.  Thanks a lot for fixing the warnings!

steveloughran commented:
suppressing the new reports would be mostly good (they are overreactions/false alarms), we may lose some real issues on the way.

zhtttylz committed:
HADOOP-19731. Fix SpotBugs warnings introduced after SpotBugs version upgrade.

steveloughran reviewed (commented):
a lot of the java changes are just removing trailing spaces. Is this needed? I knowe it's not ideal to have these, but it makes for a bigger patch and cherrypicking harder. I treat them like out-of-order imports: something I leave alone except when editing the specific lines.

spotify doesn't complain about these lines, does it?

zhtttylz commented:
> a lot of the java changes are just removing trailing spaces. Is this needed? I knowe it's not ideal to have these, but it makes for a bigger patch and cherrypicking harder. I treat them like out-of-order imports: something I leave alone except when editing the specific lines.
> 
> spotify doesn't complain about these lines, does it?

@steveloughran Thanks for the review — your point about keeping the patch small for easier cherry-picks makes sense. I’ve dropped the trailing-whitespace cleanup and introduced pattern-level filters for the new 4.9.7 rules so we reduce noise while still surfacing real issues.

The `CI: trigger SpotBugs (no-op)` commit is only there to force SpotBugs to run in pre-commit (CI skips SpotBugs when there are no `.java` changes). Once the new configuration is validated, I’ll drop that no-op commit so it doesn’t end up in the final patch.

With the current revision, the pre-commit SpotBugs summary (PR-8053/10 at https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/10/consoleText) reports `0 new` findings and only fixed issues across the modules, for example:
<details>
  <summary>SpotBugs patch summary from PR-8053/9</summary>

```
  hadoop-common-project/hadoop-common generated 0 new + 0 unchanged - 448 fixed = 0 total (was 448)

  hadoop-hdfs-project/hadoop-hdfs-client generated 0 new + 0 unchanged - 2801 fixed = 0 total (was 2801)

  hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager generated 0 new + 1 unchanged - 1298 fixed = 1 total (was 1299)
```
</details>

slfan1989 commented on hadoop-cloud-storage-project/hadoop-huaweicloud/src/test/java/org/apache/hadoop/fs/obs/OBSContract.java:
```diff
@@ -25,6 +25,7 @@
 /**
  * The contract of OBS: only enabled if the test bucket is provided.
  */
+// CI: trigger SpotBugs (no-op)
```
Do we need this change?

slfan1989 commented on hadoop-common-project/hadoop-kms/src/test/java/org/apache/hadoop/crypto/key/kms/server/KMSBenchmark.java:
```diff
@@ -614,4 +615,4 @@ public void setConf(Configuration conf) {
   }
-}
\ No newline at end of file
+}
```
Should this change be avoided as well?

zhtttylz replied:
Thank you very much for taking the time to review this. The CI result from PR-8053/10 has already validated the new SpotBugs configuration, so I will reset that temporary commit to ensure this PR no longer includes the comment-only or formatting-only changes.

slfan1989 reviewed (commented):
I generally agree with this PR, but there are a few minor issues.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   1m 13s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  0s |  |  xmllint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   7m 37s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  34m 36s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  18m 29s |  |  trunk passed with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  compile  |  19m 43s |  |  trunk passed with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  mvnsite  |  22m 20s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |  18m 31s |  |  trunk passed with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |  17m  9s |  |  trunk passed with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  shadedclient  | 171m 11s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 34s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |  26m 44s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  19m 42s |  |  the patch passed with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |  19m 42s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  21m 45s |  |  the patch passed with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |  21m 45s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |  23m 37s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |  18m 58s |  |  the patch passed with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |  18m 28s |  |  the patch passed with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  shadedclient  |  95m 29s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 24s |  |  hadoop-project in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 22s |  |  hadoop-project-dist in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 35s |  |  hadoop-minikdc in the patch passed.  |
| +1 :green_heart: |  unit  |   3m 33s |  |  hadoop-auth in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 35s |  |  hadoop-nfs in the patch passed.  |
| +1 :green_heart: |  unit  |   3m 25s |  |  hadoop-kms in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 27s |  |  hadoop-registry in the patch passed.  |
| +1 :green_heart: |  unit  |   5m 22s |  |  hadoop-hdfs-httpfs in the patch passed.  |
| +1 :green_heart: |  unit  |   3m 26s |  |  hadoop-hdfs-nfs in the patch passed.  |
| -1 :x: |  unit  | 270m 43s | [/patch-unit-hadoop-yarn-project_hadoop-yarn.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/11/artifact/out/patch-unit-hadoop-yarn-project_hadoop-yarn.txt) |  hadoop-yarn in the patch passed.  |
| +1 :green_heart: |  unit  | 142m 36s |  |  hadoop-mapreduce-client in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 25s |  |  hadoop-mapreduce-examples in the patch passed.  |
| +1 :green_heart: |  unit  | 141m 50s |  |  hadoop-mapreduce-project in the patch passed.  |
| +1 :green_heart: |  unit  |   7m  7s |  |  hadoop-streaming in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 32s |  |  hadoop-archive-logs in the patch passed.  |
| +1 :green_heart: |  unit  |   1m  5s |  |  hadoop-rumen in the patch passed.  |
| +1 :green_heart: |  unit  |  16m  0s |  |  hadoop-gridmix in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 12s |  |  hadoop-datajoin in the patch passed.  |
| -1 :x: |  unit  |   1m 41s | [/patch-unit-hadoop-tools_hadoop-sls.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/11/artifact/out/patch-unit-hadoop-tools_hadoop-sls.txt) |  hadoop-sls in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 10s |  |  hadoop-fs2img in the patch passed.  |
| +1 :green_heart: |  unit  |   3m 57s |  |  hadoop-azure in the patch passed.  |
| -1 :x: |  unit  |   3m 59s | [/patch-unit-hadoop-tools_hadoop-aws.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/11/artifact/out/patch-unit-hadoop-tools_hadoop-aws.txt) |  hadoop-aws in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 59s |  |  hadoop-aliyun in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 58s |  |  hadoop-benchmark in the patch passed.  |
| +1 :green_heart: |  unit  |   1m  5s |  |  hadoop-gcp in the patch passed.  |
| +1 :green_heart: |  unit  |   1m  2s |  |  hadoop-cos in the patch passed.  |
| +1 :green_heart: |  unit  |   1m  1s |  |  hadoop-huaweicloud in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 11s |  |  hadoop-tos in the patch passed.  |
| -1 :x: |  asflicense  |   1m 31s | [/results-asflicense.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/11/artifact/out/results-asflicense.txt) |  The patch generated 2 ASF License warnings.  |
|  |   | 963m  7s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.yarn.server.resourcemanager.TestRMHA |
|   | hadoop.yarn.server.router.subcluster.fair.TestYarnFederationWithFairScheduler |
|   | hadoop.yarn.sls.appmaster.TestAMSimulator |
|   | hadoop.fs.s3a.TestS3AUnbuffer |
|   | hadoop.fs.s3a.impl.streams.TestStreamFactories |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.52 ServerAPI=1.52 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/11/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/8053 |
| Optional Tests | dupname asflicense codespell detsecrets xmllint compile javac javadoc mvninstall mvnsite unit shadedclient |
| uname | Linux 97c6fc16637a 5.15.0-157-generic #167-Ubuntu SMP Wed Sep 17 21:35:53 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 64142937863b037b303be652975d2163a4a29fa4 |
| Default Java | Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04 |
| Multi-JDK versions | /usr/lib/jvm/java-21-openjdk-amd64:Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04 /usr/lib/jvm/java-17-openjdk-amd64:Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/11/testReport/ |
| Max. process+thread count | 1778 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-project-dist hadoop-common-project/hadoop-minikdc hadoop-common-project/hadoop-auth hadoop-common-project/hadoop-nfs hadoop-common-project/hadoop-kms hadoop-common-project/hadoop-registry hadoop-hdfs-project/hadoop-hdfs-httpfs hadoop-hdfs-project/hadoop-hdfs-nfs hadoop-yarn-project/hadoop-yarn hadoop-mapreduce-project/hadoop-mapreduce-client hadoop-mapreduce-project/hadoop-mapreduce-examples hadoop-mapreduce-project hadoop-tools/hadoop-streaming hadoop-tools/hadoop-archive-logs hadoop-tools/hadoop-rumen hadoop-tools/hadoop-gridmix hadoop-tools/hadoop-datajoin hadoop-tools/hadoop-sls hadoop-tools/hadoop-fs2img hadoop-tools/hadoop-azure hadoop-tools/hadoop-aws hadoop-tools/hadoop-aliyun hadoop-tools/hadoop-benchmark hadoop-tools/hadoop-gcp hadoop-cloud-storage-project/hadoop-cos hadoop-cloud-storage-project/hadoop-huaweicloud hadoop-cloud-storage-project/hadoop-tos U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/11/console |
| versions | git=2.25.1 maven=3.9.11 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

slfan1989 commented on dev-support/findbugs-exclude-global.xml:
```diff
@@ -0,0 +1,56 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<FindBugsFilter>
```
We need to add the Apache License notice.

zhtttylz replied:
Thanks for the feedback—I’ll make the updates right away!

zhtttylz committed:
Add ASF license header to findbugs-exclude-global.xml

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 42s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  0s |  |  xmllint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   6m 33s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  15m 10s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   8m 48s |  |  trunk passed with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  compile  |   8m 48s |  |  trunk passed with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  mvnsite  |  12m 19s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |  10m 47s |  |  trunk passed with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   9m 23s |  |  trunk passed with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  shadedclient  |  86m  8s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 19s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |  11m 44s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   8m 30s |  |  the patch passed with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |   8m 30s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   8m 41s |  |  the patch passed with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |   8m 41s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |  11m 56s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |  10m 57s |  |  the patch passed with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   9m 24s |  |  the patch passed with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  shadedclient  |  46m 57s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 13s |  |  hadoop-project in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 13s |  |  hadoop-project-dist in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 22s |  |  hadoop-minikdc in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 48s |  |  hadoop-auth in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 20s |  |  hadoop-nfs in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 39s |  |  hadoop-kms in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 50s |  |  hadoop-registry in the patch passed.  |
| +1 :green_heart: |  unit  |   3m 19s |  |  hadoop-hdfs-httpfs in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 23s |  |  hadoop-hdfs-nfs in the patch passed.  |
| -1 :x: |  unit  | 209m 57s | [/patch-unit-hadoop-yarn-project_hadoop-yarn.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/12/artifact/out/patch-unit-hadoop-yarn-project_hadoop-yarn.txt) |  hadoop-yarn in the patch passed.  |
| +1 :green_heart: |  unit  | 110m 14s |  |  hadoop-mapreduce-client in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 48s |  |  hadoop-mapreduce-examples in the patch passed.  |
| +1 :green_heart: |  unit  | 109m 32s |  |  hadoop-mapreduce-project in the patch passed.  |
| +1 :green_heart: |  unit  |   5m 18s |  |  hadoop-streaming in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 50s |  |  hadoop-archive-logs in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 38s |  |  hadoop-rumen in the patch passed.  |
| +1 :green_heart: |  unit  |  13m  3s |  |  hadoop-gridmix in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 38s |  |  hadoop-datajoin in the patch passed.  |
| -1 :x: |  unit  |   0m 52s | [/patch-unit-hadoop-tools_hadoop-sls.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/12/artifact/out/patch-unit-hadoop-tools_hadoop-sls.txt) |  hadoop-sls in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 33s |  |  hadoop-fs2img in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 25s |  |  hadoop-azure in the patch passed.  |
| -1 :x: |  unit  |   2m 15s | [/patch-unit-hadoop-tools_hadoop-aws.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/12/artifact/out/patch-unit-hadoop-tools_hadoop-aws.txt) |  hadoop-aws in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 31s |  |  hadoop-aliyun in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 28s |  |  hadoop-benchmark in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 30s |  |  hadoop-gcp in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 29s |  |  hadoop-cos in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 33s |  |  hadoop-huaweicloud in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 14s |  |  hadoop-tos in the patch passed.  |
| -1 :x: |  asflicense  |   0m 44s | [/results-asflicense.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/12/artifact/out/results-asflicense.txt) |  The patch generated 1 ASF License warnings.  |
|  |   | 640m 12s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.yarn.server.router.webapp.TestRouterWebServicesREST |
|   | hadoop.yarn.server.router.subcluster.capacity.TestYarnFederationWithCapacityScheduler |
|   | hadoop.yarn.server.resourcemanager.TestRMHA |
|   | hadoop.yarn.sls.appmaster.TestAMSimulator |
|   | hadoop.fs.s3a.impl.streams.TestStreamFactories |
|   | hadoop.fs.s3a.TestS3AUnbuffer |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.52 ServerAPI=1.52 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/12/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/8053 |
| Optional Tests | dupname asflicense codespell detsecrets xmllint compile javac javadoc mvninstall mvnsite unit shadedclient |
| uname | Linux cfbc87d89ffd 5.15.0-156-generic #166-Ubuntu SMP Sat Aug 9 00:02:46 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / d2b28189fa3f1a61d03efd5bc7da58232e54aa40 |
| Default Java | Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04 |
| Multi-JDK versions | /usr/lib/jvm/java-21-openjdk-amd64:Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04 /usr/lib/jvm/java-17-openjdk-amd64:Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/12/testReport/ |
| Max. process+thread count | 1964 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-project-dist hadoop-common-project/hadoop-minikdc hadoop-common-project/hadoop-auth hadoop-common-project/hadoop-nfs hadoop-common-project/hadoop-kms hadoop-common-project/hadoop-registry hadoop-hdfs-project/hadoop-hdfs-httpfs hadoop-hdfs-project/hadoop-hdfs-nfs hadoop-yarn-project/hadoop-yarn hadoop-mapreduce-project/hadoop-mapreduce-client hadoop-mapreduce-project/hadoop-mapreduce-examples hadoop-mapreduce-project hadoop-tools/hadoop-streaming hadoop-tools/hadoop-archive-logs hadoop-tools/hadoop-rumen hadoop-tools/hadoop-gridmix hadoop-tools/hadoop-datajoin hadoop-tools/hadoop-sls hadoop-tools/hadoop-fs2img hadoop-tools/hadoop-azure hadoop-tools/hadoop-aws hadoop-tools/hadoop-aliyun hadoop-tools/hadoop-benchmark hadoop-tools/hadoop-gcp hadoop-cloud-storage-project/hadoop-cos hadoop-cloud-storage-project/hadoop-huaweicloud hadoop-cloud-storage-project/hadoop-tos U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/12/console |
| versions | git=2.25.1 maven=3.9.11 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

steveloughran commented:
the license failures and s3a failures should all be fixed now: do a merge with trunk and non-forced update and yetus should be happy with hadoop-aws. No idea about the yarn/sls failures

slfan1989 commented:
@zhtttylz We need to rebase this branch onto trunk; the ASF license issue has already been fixed in #8098.

zhtttylz commented:
@steveloughran @slfan1989 Thanks for the reminder. I’ve rebased HADOOP-19731 onto the latest trunk (with the ASF license fix from #8098) and pushed the updated branch so Yetus can rerun. Really appreciate your help here.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  21m 38s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  0s |  |  xmllint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   9m 53s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  35m 57s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  18m 58s |  |  trunk passed with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  compile  |  19m 17s |  |  trunk passed with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  mvnsite  |  24m 23s |  |  trunk passed  |
| -1 :x: |  javadoc  |   0m 38s | [/branch-javadoc-hadoop-tools_hadoop-azure-jdkUbuntu-21.0.7+6-Ubuntu-0ubuntu120.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/13/artifact/out/branch-javadoc-hadoop-tools_hadoop-azure-jdkUbuntu-21.0.7+6-Ubuntu-0ubuntu120.04.txt) |  hadoop-azure in trunk failed with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04.  |
| -1 :x: |  javadoc  |   0m 35s | [/branch-javadoc-hadoop-tools_hadoop-azure-jdkUbuntu-17.0.15+6-Ubuntu-0ubuntu120.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/13/artifact/out/branch-javadoc-hadoop-tools_hadoop-azure-jdkUbuntu-17.0.15+6-Ubuntu-0ubuntu120.04.txt) |  hadoop-azure in trunk failed with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04.  |
| +1 :green_heart: |  shadedclient  | 176m 31s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 35s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |  23m 18s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  18m 10s |  |  the patch passed with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |  18m 10s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  19m 18s |  |  the patch passed with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |  19m 18s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |  24m 17s |  |  the patch passed  |
| -1 :x: |  javadoc  |   0m 37s | [/patch-javadoc-hadoop-tools_hadoop-azure-jdkUbuntu-21.0.7+6-Ubuntu-0ubuntu120.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/13/artifact/out/patch-javadoc-hadoop-tools_hadoop-azure-jdkUbuntu-21.0.7+6-Ubuntu-0ubuntu120.04.txt) |  hadoop-azure in the patch failed with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04.  |
| -1 :x: |  javadoc  |   0m 38s | [/patch-javadoc-hadoop-tools_hadoop-azure-jdkUbuntu-17.0.15+6-Ubuntu-0ubuntu120.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/13/artifact/out/patch-javadoc-hadoop-tools_hadoop-azure-jdkUbuntu-17.0.15+6-Ubuntu-0ubuntu120.04.txt) |  hadoop-azure in the patch failed with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04.  |
| +1 :green_heart: |  shadedclient  |  94m 15s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 24s |  |  hadoop-project in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 25s |  |  hadoop-project-dist in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 35s |  |  hadoop-minikdc in the patch passed.  |
| +1 :green_heart: |  unit  |   3m 28s |  |  hadoop-auth in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 33s |  |  hadoop-nfs in the patch passed.  |
| +1 :green_heart: |  unit  |   3m 16s |  |  hadoop-kms in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 19s |  |  hadoop-registry in the patch passed.  |
| +1 :green_heart: |  unit  |   5m  0s |  |  hadoop-hdfs-httpfs in the patch passed.  |
| +1 :green_heart: |  unit  |   3m 22s |  |  hadoop-hdfs-nfs in the patch passed.  |
| -1 :x: |  unit  | 263m 57s | [/patch-unit-hadoop-yarn-project_hadoop-yarn.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/13/artifact/out/patch-unit-hadoop-yarn-project_hadoop-yarn.txt) |  hadoop-yarn in the patch passed.  |
| +1 :green_heart: |  unit  | 141m 33s |  |  hadoop-mapreduce-client in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 27s |  |  hadoop-mapreduce-examples in the patch passed.  |
| +1 :green_heart: |  unit  | 140m 32s |  |  hadoop-mapreduce-project in the patch passed.  |
| +1 :green_heart: |  unit  |   7m 23s |  |  hadoop-streaming in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 37s |  |  hadoop-archive-logs in the patch passed.  |
| +1 :green_heart: |  unit  |   1m  7s |  |  hadoop-rumen in the patch passed.  |
| +1 :green_heart: |  unit  |  21m 56s |  |  hadoop-gridmix in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 16s |  |  hadoop-datajoin in the patch passed.  |
| -1 :x: |  unit  |   2m 57s | [/patch-unit-hadoop-tools_hadoop-sls.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/13/artifact/out/patch-unit-hadoop-tools_hadoop-sls.txt) |  hadoop-sls in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 43s |  |  hadoop-fs2img in the patch passed.  |
| +1 :green_heart: |  unit  |   4m  5s |  |  hadoop-azure in the patch passed.  |
| +1 :green_heart: |  unit  |   4m  8s |  |  hadoop-aws in the patch passed.  |
| +1 :green_heart: |  unit  |   1m  9s |  |  hadoop-aliyun in the patch passed.  |
| +1 :green_heart: |  unit  |   1m  6s |  |  hadoop-benchmark in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 13s |  |  hadoop-gcp in the patch passed.  |
| +1 :green_heart: |  unit  |   1m  6s |  |  hadoop-cos in the patch passed.  |
| +1 :green_heart: |  unit  |   1m  4s |  |  hadoop-huaweicloud in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 15s |  |  hadoop-tos in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m 41s |  |  The patch does not generate ASF License warnings.  |
|  |   | 980m 43s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.yarn.service.TestYarnNativeServices |
|   | hadoop.yarn.server.resourcemanager.TestRMHA |
|   | hadoop.yarn.server.router.subcluster.fair.TestYarnFederationWithFairScheduler |
|   | hadoop.yarn.sls.appmaster.TestAMSimulator |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.52 ServerAPI=1.52 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/13/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/8053 |
| Optional Tests | dupname asflicense codespell detsecrets xmllint compile javac javadoc mvninstall mvnsite unit shadedclient |
| uname | Linux cc4aa6a57dbe 5.15.0-157-generic #167-Ubuntu SMP Wed Sep 17 21:35:53 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / a076fdacb7eee524059945f9b795bfbb7999866a |
| Default Java | Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04 |
| Multi-JDK versions | /usr/lib/jvm/java-21-openjdk-amd64:Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04 /usr/lib/jvm/java-17-openjdk-amd64:Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/13/testReport/ |
| Max. process+thread count | 1784 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-project-dist hadoop-common-project/hadoop-minikdc hadoop-common-project/hadoop-auth hadoop-common-project/hadoop-nfs hadoop-common-project/hadoop-kms hadoop-common-project/hadoop-registry hadoop-hdfs-project/hadoop-hdfs-httpfs hadoop-hdfs-project/hadoop-hdfs-nfs hadoop-yarn-project/hadoop-yarn hadoop-mapreduce-project/hadoop-mapreduce-client hadoop-mapreduce-project/hadoop-mapreduce-examples hadoop-mapreduce-project hadoop-tools/hadoop-streaming hadoop-tools/hadoop-archive-logs hadoop-tools/hadoop-rumen hadoop-tools/hadoop-gridmix hadoop-tools/hadoop-datajoin hadoop-tools/hadoop-sls hadoop-tools/hadoop-fs2img hadoop-tools/hadoop-azure hadoop-tools/hadoop-aws hadoop-tools/hadoop-aliyun hadoop-tools/hadoop-benchmark hadoop-tools/hadoop-gcp hadoop-cloud-storage-project/hadoop-cos hadoop-cloud-storage-project/hadoop-huaweicloud hadoop-cloud-storage-project/hadoop-tos U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8053/13/console |
| versions | git=2.25.1 maven=3.9.11 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

slfan1989 commented:
@zhtttylz I’m going to merge this PR, as the JavaDoc error is unrelated to these changes. Many thanks to @steveloughran and @szetszwo for reviewing the code.
